### PR TITLE
Fix some bugs with the cfb1 bitsize handling

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

**Comment by kroeckx** [comment]
This looks like something that could use test cases.

**Comment by bernd-edlinger** [comment]
This is difficult to test because the failing code path is only entered when the input length
is larger than MAXBITCHUNK / EVP_MAXBLOCKSIZE.

for AES / camellia we have:
 # define MAXBITCHUNK ((size_t)1<<(sizeof(size_t)*8-4))

for most other ciphers we have:
#define EVP_MAXCHUNK ((size_t)1<<(sizeof(long)*8-2))

If that constant is lowered to say 256 bytes you can see 20-test_enc and
20-test_enc_more.t failing without this patch.

**Comment by bernd-edlinger** [comment]
However, I think EVP_MAXCHUNK is too high if shifted up by 3 bits it will overflow.....

**Comment by bernd-edlinger** [comment]
Disregard the last comment, the chunk size is shifted down by 3 bits.
so the input length needs to be larger than EVP_MAXBLOCKSIZE/8
to cause a problem.

**Comment by bernd-edlinger** [comment]
So on a 32-bit target encrypting more than 2^27 bytes = 128 MB is bound to fail.

File: crypto/include/internal/evp_int.h [hunk 1]
```diff
@@ -214,7 +214,7 @@ static int cname##_cfb##cbits##_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
         cprefix##_cfb##cbits##_encrypt(in, out, (long) \
             ((cbits == 1) \
                 && !EVP_CIPHER_CTX_test_flags(ctx, EVP_CIPH_FLAG_LENGTH_BITS) \
-                ? inl*8 : inl), \
+                ? chunk*8 : chunk), \
```
Comment (dot-asm) [inline crypto/include/internal/evp_int.h]: Hmmm... Originally it was chunk...

**Comment by dot-asm** [comment]
2^28, right? It's unreasonable to mount such test. So I'd say it's ready as is.

[Just in case, since it was mentioned, in case one wonders that is. What's EVP_MAXCHUNK and why does it look the way it does. It's there to reconcile the fact that EVP accepts len as size_t, while *some* of lower level subroutines as [signed] long :-(]

**Comment by bernd-edlinger** [comment]
Merged to master and 1.1.0 so far.
Thanks!

**Pull request closed without merging**